### PR TITLE
Modify GPSD Application to support SNMP Extend. Update Docs

### PR DESCRIPTION
**Pull request by mikecentola**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/polling/applications/gpsd.inc.php [hunk 1]
```diff
@@ -24,45 +24,135 @@
...
+        foreach ($check_fields as $field) {
+            if (!empty($gpsd_parsed[$field])) {
+                $fields[$field] = $gpsd_parsed[$field];
+            }
+        }
+    } else {
+        // Use json_app_get to grab JSON formatted GPSD data
+        try {
+            $gpsd = json_app_get($device, $name);
+        } catch (JsonAppParsingFailedException $e) {
```
Comment (murrant) [inline includes/polling/applications/gpsd.inc.php]: You need to either add a use statement for this or the full namespace.

File: includes/polling/applications/gpsd.inc.php [hunk 2]
```diff
@@ -24,45 +24,135 @@
...
+            $legacy = $e->getOutput();
+
+            $gpsd = array(
+                data => array(),
+            );
+
+            list ($gpsd['data']['mode'], $gpsd['data']['hdop'], $gpsd['data']['vdop'],
+                $gpsd['data']['latitude'], $gpsd['data']['longitude'], $gpsd['data']['altitude'],
+                $gpsd['data']['satellites'], $gpsd['data']['satellites_used']) = explode("\n", $legacy);
+        } catch (JsonAppException $e) {
```
Comment (murrant) [inline includes/polling/applications/gpsd.inc.php]: Same here.

File: includes/polling/applications/gpsd.inc.php [hunk 3]
```diff
@@ -24,45 +24,135 @@
...
     }
 
+    // Generate RRD Def
+
     $rrd_name = array('app', $name, $app_id);
     $rrd_def = RrdDefinition::make()
         ->addDataset('mode', 'GAUGE', 0, 4)
         ->addDataset('hdop', 'GAUGE', 0, 100)
         ->addDataset('vdop', 'GAUGE', 0, 100)
+        //->addDataset('latitude', 'GAUGE', -90, 90)
```
Comment (murrant) [inline includes/polling/applications/gpsd.inc.php]: Don't leave comment code in, just remove it.

**Comment by mikecentola** [comment]
Updated the file with changes.

**Pull request merged**